docs/deploying.md

```diff
@@ -20,7 +20,7 @@ Test the output looks visually OK before / after with a few sample PDFs. You sho
 
```

Comment:
Thanks for spotting - I've amended the URL 👍.

This is a tricky one: these are really visual regression tests; if we want to keep them, then I'd like to make them proper automated tests and use a library to compare the before / after images automatically. However, I'm unsure if we want to keep them: doing these tests manually hasn't detected any issues since we introduced them, except in [the library to convert PDFs to PNGs](https://github.com/alphagov/notifications-template-preview/pull/591#issuecomment-979284671), which isn't critical. It felt reassuring to have them.

Historically all the issues we've had are with DVLA being unable to print a PDF due to a quirk of it's internals - I'm not aware of any visual hints of this, or any visual issues with PDFs in general. That said, when we're making a change like https://github.com/alphagov/notifications-template-preview/pull/632 and passing the entire PDF through a whole new mill, I do think we should be checking it hasn't done something awful to the output e.g. made it greyscale, deleted all the images, etc.

Short version:

- Do visual checks actually prevent scary issues at DVLA? No, I don't think so.
- Should we be visually checking PDFs when we make big changes? Probably.

Possible action:

- Move this guidance to a separate "visual testing" doc if we think it's discretionary.
- Suggest visual regression as an alternative approach on [this card](https://www.pivotaltracker.com/story/show/178746414).

What do you think?

Reply:
Yeah I definitely agree with you on those points. I don't think visual regression checks tie in with the thngs that DVLA struggle with but they still provide some value to us. We've historically had some issues where some PDFs we'd end up removing the logos when converting to CMYK IIRC? Others in the team might remember more. Not quite sure what it was - think we ended up bumping some libraries and that helped fix the issue. An easy way to test visually then might have helped.

Also, semi relatedly presumably flask shell is still a bit of a faff cos you need to open up a docker shell, run the stuff, work out where the files are, get them out with `docker cp`, compare them locally, etc. I think I ended up just writing a script using the python client to go via API at the time.

Reply:
as to your actions, i like moving it to a separate section. maybe reference that in this main bit and say "if you're worried your changes might impact pdf generation, consider running a manual visual regression test [ link to lower section ]"